flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/overlays/HadoopUserOverlay.java

```diff
@@ -57,24 +57,58 @@ public void configure(ContainerSpecification container) throws IOException {
 	}
 
 	public static Builder newBuilder() {
-		return new Builder();
+		// First check if we have Hadoop in the ClassPath. If not, we simply don't do anything.
+		try {
+			Class.forName(
+				"org.apache.hadoop.security.UserGroupInformation",
+				false,
+				HadoopUserOverlay.class.getClassLoader());
+		} catch (ClassNotFoundException e) {
+			LOG.info("Cannot create Hadoop User Overlay because Hadoop cannot be found in the Classpath.");
+			return new NoOpHadoopUserOverlayBuilder();
+		}
+
+		return new HadoopUserOverlayBuilder();
 	}
 
 	/**
 	 * A builder for the {@link HadoopUserOverlay}.
 	 */
-	public static class Builder {
+	public interface Builder {
+		/**
+		 * Configures the overlay using the current Hadoop user information (from {@link UserGroupInformation}).
+		 */
+		Builder fromEnvironment(Configuration globalConfiguration) throws IOException;
+
+		HadoopUserOverlay build();
+	}
+
+	/**
+	 * A builder for {@link HadoopUserOverlay} for when hadoop doesn't exist on the classpath
+	 */
+	public static class NoOpHadoopUserOverlayBuilder implements Builder {
```

Comment:
what's the meaning of "Op" in class name `NoOpHadoopUserOverlayBuilder `, maybe `NoHadoopUserOverlayBuilder` or `EmptyHadoopUserOverlayBuilder` looks better, just personal opinion.

Reply:
I was going for `no-op` here, but I agree that Empty might convey a better meaning